enhancements/api-review/azure_user_defined_tags.md

```diff
@@ -0,0 +1,310 @@
+---
+title: apply-user-defined-tags-to-all-azure-resources-created-by-openShift
+authors:
+  - "@bhb"
+reviewers:
+  - @patrickdillon
+  - @sdodson
+  - @jerpeter1
+  - @Miciah
+  - @sinnykumari
+  - @dmage
+  - @staebler
+  - @tkashem
+  - @tjungblu
+approvers:
+  - @sdodson
+  - @jerpeter1
+  - @bparees
+api-approvers:
+  - @joelspeed
+creation-date: 2022-07-12
+last-updated: 2022-07-12
+tracking-link:
+  - https://issues.redhat.com/browse/OCPPLAN-8155
+see-also:
+  - "enhancements/api-review/custom-tags-aws.md"
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# Apply user defined tags to all Azure resources created by OpenShift
+
+## Summary
+
+This enhancement describes the proposal to allow an administrator of Openshift to 
+have the ability to apply user defined tags to those resources created by Openshift 
+in Azure.
+
+## Motivation
+
+Motivations include but are not limited to:
+
+- Allow admin, compliance, and security teams to keep track of assets and objects 
+  created by OpenShift in Azure.
+
+### User Stories
+
+- As an openshift administrator, I want to have tags added to all resources created 
+  in Azure by Openshift.
+- As an openshift administrator, I want to restrict access granted to Openshift specific account.
+
+### Goals
+
+- The administrator or service (in the case of Managed OpenShift) installing OpenShift 
+  can configure allowed number of user-defined tags in the Openshift installer generated
+  install config, which is referred and applied by installer and the in-cluster operators
+  on the the Azure resources during creation.
+- Tags must be applied at creation time, in an atomic operation. It isn't acceptable 
+  to create an object and, after some period of time, apply the tags post-creation.
```

Comment:
Is there a list of objects where this cannot be met?  I haven't looked at Azure specifically but there are some resources in other cloud providers that cannot be tagged atomically, it requires an update to the resource after creation.

Reply:
The Azure resources which are created by installer using terraform API's, all have support for adding tags during creation. Didn't come across any resource, those created by installer, requiring additional request for adding tags.